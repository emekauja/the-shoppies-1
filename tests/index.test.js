const request = require('supertest')
const nock = require('nock')

const app = require('../index')

describe('Endpoints', () => {
  test('return OMDb movies', async () => {
    const scope = nock('http://www.omdbapi.com')
      .get('/?apikey=12345&type=movie&s=vertigo')
      .reply(200, {
        "Search": [
          {
            "Title": "Vertigo",
            "Year": "1958",
            "imdbID": "tt0052357",
            "Type": "movie",
            "Poster": "https://m.media-amazon.com/images/M/MV5BYTE4ODEwZDUtNDFjOC00NjAxLWEzYTQtYTI1NGVmZmFlNjdiL2ltYWdlL2ltYWdlXkEyXkFqcGdeQXVyNjc1NTYyMjg@._V1_SX300.jpg"
          },
        ],
      })

    const res = await request(app)
      .get('/api/getMovies')
      .query({ search: 'vertigo' })

    expect(res.statusCode).toEqual(200)
    expect(res.body).toMatchSnapshot()
  })
})
