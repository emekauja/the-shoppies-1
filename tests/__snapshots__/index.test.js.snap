// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Endpoints return OMDb movies 1`] = `
Object {
  "Search": Array [
    Object {
      "Poster": "https://m.media-amazon.com/images/M/MV5BYTE4ODEwZDUtNDFjOC00NjAxLWEzYTQtYTI1NGVmZmFlNjdiL2ltYWdlL2ltYWdlXkEyXkFqcGdeQXVyNjc1NTYyMjg@._V1_SX300.jpg",
      "Title": "Vertigo",
      "Type": "movie",
      "Year": "1958",
      "imdbID": "tt0052357",
    },
  ],
}
`;
